lib/ohai/plugins/linux/virtualization.rb

```diff
@@ -150,6 +150,17 @@ def docker_exists?
       end
     end
 
+    # Detect Hyper-V guest and the hostname of the host
+    if File.exist?("/var/lib/hyperv/.kvp_pool_3")
+      logger.trace("Plugin Virtualization: /var/lib/hyperv/.kvp_pool_3 contains string indicating Hyper-V guest")
+      data = File.read("/var/lib/hyperv/.kvp_pool_3")
+      hyperv_host = data[/\HostName(.*?)HostingSystemEditionId/, 1].scan(/[[:print:]]/).join.downcase
+      virtualization[:system] = "hyperv"
+      virtualization[:role] = "guest"
+      virtualization[:systems][:hyperv] = "guest"
+      virtualization[:host] = hyperv_host
```

Comment:
I'm not sure how I feel about this attribute name. It seems a bit ambiguous. How would you feel about hypervisor_host instead? I'm open to other ideas as well.

Reply:
That's a good idea, I think "hypervisor_host" is better than "host". That's more understandable and less confusing. :blush: